### DISCUSSION TITLE
Руководство ПЛИ 3.10 Документируемое событие <documentationOf>

### DISCUSSION THREAD

**Discussion by eugenekogan-spb** [fhir-ru/core#278] [Лабораторная РГ]
Элемент documentationOf описывает документированное событие и содержит: дату и время формирования протокола лабораторного исследования, которые указываются в элементе effectiveTime, а также медицинский персонал, участвовавший в заборе, обработке биоматериала и проведении собственно лабораторных исследований.
Все лица, участвовавшие в проведении исследования, указываются в элементах performer. Участник может быть отмечен как непосредственный исполнитель документируемого события, либо как вспомогательный или ассистент, что задаётся в атрибуте performer/@typeCode. Значения берутся из справочника HL7ParticipationType (OID 2.16.840.1.113883.5.90): "PPRF" обозначает непосредственного исполнителя, "SPRF" – вспомогательного\ассистента.
В структурированном теле электронного документа могут указываться только идентификаторы исполнителей, которые нужно будет использовать в качестве ссылки для подгрузки информации, указанной в данном разделе.

**Comment by eugenekogan-spb** [comment]
<img width="770" alt="image" src="https://github.com/user-attachments/assets/3a44127a-4a5a-4bd0-a7ba-e3f30eb481c6">
<img width="770" alt="image" src="https://github.com/user-attachments/assets/fbc36d2f-c92d-4be7-8da4-8e662b86a0cb">

**Reply by eugenekogan-spb** [reply]
Для передачи данной информации добавляем следующие элементы

```
//DiagnosticReport

Instance:   diagnostic-report-pli
InstanceOf: DiagnosticReport
Title: "Lab DiagnosticReport Instance PLI"
Description: "Результат лабораторного от 31.10.2022"
Usage: #example

* status = #final

* category = http://terminology.hl7.org/CodeSystem/v2-0074#LAB

* code
  * text = "Какое-то лабораторное исследование"

* subject = Reference (patient-maria-antuanetta)

* effectiveDateTime = "2022-10-31T21:50:00+04:00" 

* performer = Reference (practitionerrole-lab-doctor-helix-altay)

//ServiceRequest

InstanceOf: ServiceRequest
* encounter = Reference (encounter-outpatient)

//Encounter: outpatient

Instance:   encounter-outpatient
InstanceOf: Encounter
Title: "Lab Encounter Instance Outpatient"
Description: "Амбулаторный случай от 30.10.2022"
Usage: #example

* type[+]
  * text = "Планово"
  * coding
    * system = "Core_Cs_Nsi_Service-Form"
    * code = #1

* type[+]
  * text = "первичная врачебная медико-санитарная помощь"
  * coding
    * system = "Core_Cs_Nsi_Service-Type"
    * code = #2

* type[+]
  * text = "амбулаторно, в том числе на дому при вызове медицинского работника"
  * coding
    * system = "Core_Cs_Nsi_Service-Condition"
    * code = #2

//Формы оказания медицинской помощи
CodeSystem: Core_Cs_Nsi_Service_Form
Id: core-cs-nsi-service-form
* #1 "планово"

//Виды оказания медицинской помощи
CodeSystem: Core_Cs_Nsi_Service_Type
Id: core-cs-nsi-service-type
* #2 "первичная врачебная медико-санитарная помощь"

//Условия оказания медицинской помощи
CodeSystem: Core_Cs_Nsi_Service_Condition
Id: core-cs-nsi-service-condition
* #2 "амбулаторно, в том числе на дому при вызове медицинского работника"

```

**Comment by eugenekogan-spb** [comment]
3.10.1. Документируемое событие, его тип и дата

У1-21: Элемент ClinicalDocument ОБЯЗАН иметь один [1..1] элемент
documentationOf, содержащий сведения о документируемом событии, который: 1. ОБЯЗАН иметь один [1..1] элемент serviceEvent, который:
1. ОБЯЗАН иметь один [1..1] элемент code (атрибут @code должен иметь значение "4", атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.99.2.726", атрибут @codeSystemName должен иметь значение "Типы документированных событий", атрибут @displayName должен иметь значение "Лабораторное исследование", атрибут @codeSystemVersion должен быть представлен с не пустым значением), содержащий тип документируемого события;
2. ОБЯЗАН иметь один [1..1] элемент effectiveTime (атрибут @value должен быть представлен с не пустым значением), содержащий дату формирования протокола лабораторного исследования;

**Reply by eugenekogan-spb** [reply]
Комментарий: в НСИ МЗ РФ [нет справочника с OID 1.2.643.5.1.13.13.99.2.726](https://fnsi2.test.egisz.rosminzdrav.ru/dictionaries?&page=1&size=20&query=1.2.643.5.1.13.13.99.2.726).

**Reply by eugenekogan-spb** [reply]
documentationOf передается ресурсом DiagnosticReport, в котором согласно ранее предложенному решению code означает оказанную медицинскую услугу и заполняется по справочнику 1.2.643.5.1.13.13.11.1070

```
Instance:   diagnostic-report-pli
InstanceOf: DiagnosticReport
Title: "Lab DiagnosticReport Instance PLI"
Description: "Результат лабораторного от 31.10.2022"
Usage: #example

* status = #final

* category = http://terminology.hl7.org/CodeSystem/v2-0074#LAB

* code
  * text = "Какое-то лабораторное исследование"

* subject = Reference (patient-maria-antuanetta)

* encounter = Reference (encounter-outpatient)

* effectiveDateTime = "2022-10-31T21:50:00+04:00" 

```

**Reply by maximalexeev** [reply]
В категорию я бы лучше завез вид исследования по 1.2.643.5.1.13.13.11.1117

**Reply by eugenekogan-spb** [reply]
Принято на встрече 12.11.24.

**Reply by eugenekogan-spb** [reply]
Тогда получаются множественные категории. Категорирование по FHIR можно оставить для совместимости, чтобы передать результаты в зарубежную FHIR-систему. И дополнительно, ставить категорию по ФСЛИ.

Вопрос о том, что записывать в code, остался не решенным. На встрече 12.11.24 принято, что DiagnosticReport можно создавать один на группу результатов, которые будут оплачиваться несколькими услугами. Поэтому услуги кодируются в расширении.

```
Instance:   diagnostic-report-pli
InstanceOf: DiagnosticReport
Title: "Lab DiagnosticReport Instance PLI"
Description: "Результат лабораторного от 31.10.2022"
Usage: #example

* status = #final

* category[+] = http://terminology.hl7.org/CodeSystem/v2-0074#LAB
* category[+]
  * text = "Биохимические исследования"
  * coding
    * system = "1.2.643.5.1.13.13.11.1117"
    * code = #201
* code
  * text = "Какое-то лабораторное исследование"

* subject = Reference (patient-maria-antuanetta)

* encounter = Reference (encounter-outpatient)

* effectiveDateTime = "2022-10-31T21:50:00+04:00" 

```

**Reply by eugenekogan-spb** [reply]
Видимо, следуя логике руководства, необходимо оформить справочник типов документируемых событий, несмотря на его отсутствие в НСИ.
```
* code
  * text = "Лабораторное исследование"
  * coding
    * system = "Core_Cs_Nsi_Event_Type"
    * code = #4
```

!!! Решение о создании справочника не обсуждалось

**Comment by eugenekogan-spb** [comment]
3.10.2. Форма МП, вид МП, условия оказания МП.

3.ДОЛЖЕН иметь один [1..1] элемент medService:serviceForm (атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.11.1551", атрибут @codeSystemName должен иметь значение "Формы оказания медицинской помощи", атрибуты @code, @codeSystemVersion, @displayName должны быть представлены с не пустыми значениями), содержащий форму оказания медицинской помощи;
4.ДОЛЖЕН иметь один [1..1] элемент medService:serviceType (атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.11.1034", атрибут @codeSystemName должен иметь значение "Виды медицинской помощи", атрибуты @code, @codeSystemVersion, @displayName должны быть представлены с не пустыми значениями), содержащий вид медицинской помощи;
5.ДОЛЖЕН иметь один [1..1] элемент medService:serviceCond (атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.99.2.322", атрибут @codeSystemName должен иметь значение "Условия оказания медицинской помощи", атрибуты @code, @codeSystemVersion, @displayName должны быть представлены с не пустыми значениями), содержащий условия оказания медицинской помощи;

**Reply by eugenekogan-spb** [reply]
Форма, вид и условия оказания медицинской помощи (откладываем в сторону вопрос о цели их передачи в составе протокола лабораторного исследования) относятся к случаю медицинского обслуживания, в рамках которого было выполнено лабораторное исследование

```
Instance:   encounter-outpatient
InstanceOf: Encounter
Title: "Lab Encounter Instance Outpatient"
Description: "Амбулаторный случай от 30.10.2022"
Usage: #example

* type[+]
  * text = "Планово"
  * coding
    * system = "Core_Cs_Nsi_Service-Form"
    * code = #1

* type[+]
  * text = "первичная врачебная медико-санитарная помощь"
  * coding
    * system = "Core_Cs_Nsi_Service-Type"
    * code = #2

* type[+]
  * text = "амбулаторно, в том числе на дому при вызове медицинского работника"
  * coding
    * system = "Core_Cs_Nsi_Service-Condition"
    * code = #2

//Формы оказания медицинской помощи
CodeSystem: Core_Cs_Nsi_Service_Form
Id: core-cs-nsi-service-form
* #1 "планово"

//Виды оказания медицинской помощи
CodeSystem: Core_Cs_Nsi_Service_Type
Id: core-cs-nsi-service-type
* #2 "первичная врачебная медико-санитарная помощь"

//Условия оказания медицинской помощи
CodeSystem: Core_Cs_Nsi_Service_Condition
Id: core-cs-nsi-service-condition
* #2 "амбулаторно, в том числе на дому при вызове медицинского работника"

```

**Reply by eugenekogan-spb** [reply]
Видимо, в тексте выше неточность. Это случай обслуживания, в рамках которого было заказано исследование. Поэтому Encounter надо включить в ServiceRequest.

**Comment by eugenekogan-spb** [comment]
3.10.3. исполнитель
6. ОБЯЗАН иметь хотя бы один [1..*] элемент performer (атрибут @typeCode должен быть представлен с не пустым значением), содержащий сведения об участнике документируемого события. ЕСЛИ описывается непосредственный исполнитель, ТО атрибут performer/@typeCode должен иметь значение "PPRF"; ЕСЛИ описывается вспомогательный исполнитель или ассистент, ТО атрибут performer/@typeCode должен иметь значение "SPRF". Данный элемент:
1. ОБЯЗАН иметь один [1..1] элемент assignedEntity, который:
1.ДОЛЖЕН иметь один [1..1] элемент id (атрибуты @root, @extension должны быть представлены с не пустыми значениями; атрибут @root должен быть сформирован по правилу формирования идентификаторов персонала, т.е. "OID_медицинской_организации.100.НомерМИС.НомерЭкзМИС.70", атрибут @extension должен соответствовать идентификатору персонала), содержащий уникальный идентификатор участника документируемого события в МИС;
2.ОБЯЗАН иметь один [1..1] элемент code (атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.11.1002", атрибут @codeSystemName должен иметь значение "Должности медицинских и фармацевтических работников", атрибуты @code, @codeSystemVersion, @displayName должны быть представлены с не пустыми значениями), содержащий код должности участника документируемого события;
3.МОЖЕТ иметь один [0..1] элемент telecom...
4.ОБЯЗАН иметь один [1..1] элемент assignedPerson...

**Reply by eugenekogan-spb** [reply]
сам участник передается элементом DiagnosticReport.performer 
```
* performer = Reference (practitionerrole-lab-doctor-helix-altay)

```

однако его роль в проведении исследования / написании протокола ("PPRF" или "SPRF") не входит в стандартный состав атрибутов DiagnosticReport. Указание роли из такого списка предусмотрено в Encounter.

**Reply by eugenekogan-spb** [reply]
!!! Решение, как указывать роль исполнителя (PPRF,SPRF в примере - не обсуждалось.
Возможно, следует смоделировать "документируемое событие" - лабораторное исследование, как самостоятельный encounter, и в его описании указать роли участников.

**Reply by eugenekogan-spb** [reply]
На встрече 17.01.25 решили: для подробного описания команды участников исследования с их ролями следует описать исполнителя как CareTeam
```
* performer = Reference (careteam-laboratory-staff)

Instance: careteam-laboratory-staff
InstanceOf: CareTeam
Usage: #inline

* participant[+]
  * member = Reference (practitionerrole-stepanova-author)
* participant[+]
  * member = Reference (practitionerrole-ivanov-authenticator)
* participant[+]
  * member = Reference (practitionerrole-arhipova-assistant)
```

**Reply by eugenekogan-spb** [reply]
На встрече 21.01.25 определили, что нет справочника ролей участников лабораторного исследования. Поэтому достаточно передать text, а также код по системе кодирования http://terminology.hl7.org/CodeSystem/v3-ParticipationType - только PPRF и SPRF (для профилирования потребуется набор значений).

```
Instance: careteam-laboratory-staff
InstanceOf: CareTeam
Usage: #inline

* participant[+]
  * role
    * text = "автор"
    * coding = http://terminology.hl7.org/CodeSystem/v3-ParticipationType#PPRF
  * member = Reference (practitionerrole-stepanova-author)
* participant[+]
    * text = "авторизовал результаты"
    * coding = http://terminology.hl7.org/CodeSystem/v3-ParticipationType#SPRF
  * member = Reference (practitionerrole-ivanov-authenticator)
* participant[+]
    * text = "взятие образца"
    * coding = http://terminology.hl7.org/CodeSystem/v3-ParticipationType#SPRF
  * member = Reference (practitionerrole-arhipova-assistant)
```
